crates/runtime/src/flight/actions/datasets.rs

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use tonic::Status;
+
+use crate::{datafusion::Error, flight::Service};
+
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct ActionAcceleratedDatasetRefreshRequest {
+    #[prost(string, tag = "1")]
+    pub dataset_name: ::prost::alloc::string::String,
+}
+
+pub(crate) async fn do_action_accelerated_dataset_refresh(
+    flight_svc: &Service,
+    cmd: ActionAcceleratedDatasetRefreshRequest,
+) -> Result<(), Status> {
+    tracing::trace!("do_action_accelerated_dataset_refresh: {cmd:?}");
+
+    let dataset_name = &cmd.dataset_name;
+
+    match flight_svc.datafusion.refresh_table(&cmd.dataset_name).await {
```

Comment:
This should probably take a `TableReference` not a string, since this wouldn't work for datasets configured with a schema in the name.

Reply:
And then that would mean we would need to call TableReference::parse_str on `cmd.dataset_name`

Reply:
Verified names with schema support - this works correct as `table_provider` method accept `table_ref: impl Into<TableReference>` and there is special helper that automatically parses string to TableReference so manual string to TableReference conversion is not required. Also tested to make sure it works, will include table name with schema in tests as well
```
impl<'a> From<&'a str> for TableReference {
    fn from(s: &'a str) -> Self {
        Self::parse_str(s)
    }
}
...
```